modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -125,82 +144,41 @@ def exploit
     end
     print_good 'Passed all initial checks for exploit'
 
-
     xdisplay = datastore['Xdisplay']
-    in_shell = datastore['INSHELL']
-    payload_path = datastore['PAYLOAD_LOC']
-    @pscript = "#{datastore['WritableDir']}/.session-#{rand_text_alphanumeric 5..10}"
-    payload_c = "#{@pscript}.c"
-
-    #Needed to delete some files in new session
-    def on_new_session(session)
-      super
-      session.shell_command("cat #{@pscript}.b > /etc/crontab")
-      session.shell_command("rm -f #{@pscript}.b")
-      session.shell_command('rm -f /etc/crontab.old')
-      print_status "Crontab restored and files deleted, enjoy your shell"
-    end
+    pscript = "#{datastore['WritableDir']}/.session-#{rand_text_alphanumeric 5..10}"
+    @clean_up = "/bin/cat #{pscript}.b > /etc/crontab ; /bin/rm -f #{pscript}* /etc/crontab.old"
```

Comment:
`/bin/rm -f #{pscript}*`

Please don't. This is [asking for trouble](https://www.theregister.co.uk/2015/01/17/scary_code_of_the_week_steam_cleans_linux_pcs/).

Reply:
It is initialized to working dir + .session-(rand).  Would already need to be run by root in the crontab.  You ok if I take out the -f ?

Reply:
Don't use `*`. Specify the path.

Reply:
For example, what if `WritableDir` was `/home/heathen/i use spaces in my file and directory names because i'm a degenerate psychopath`